src/utils.cpp

```diff
@@ -226,4 +230,257 @@ namespace diskann {
                   << std::endl;
   }
 
+ #ifdef EXEC_ENV_OLS
+  void get_bin_metadata(AlignedFileReader& reader, size_t& npts, size_t& ndim,
```

Comment:
Do you prefer to replicate most of the load code between EXEC_ENV_OLS and other environments? Or is there a way to share most of the code and avoid replication?

Reply:
Apart from a few assignments, there won't be much code duplication